Frameworks/CoreText/DWriteWrapper.mm

```diff
@@ -493,6 +507,42 @@ static void __GetGlyphRunDetails(_CTTypesetter* ts, CFRange range, CGRect frameS
                 line->_width += glyphRunDetails._dwriteGlyphRun[i].glyphAdvances[index];
             }
 
+            DWRITE_GLYPH_METRICS glyphMetrics[run->_dwriteGlyphRun.glyphCount];
+            THROW_IF_FAILED(run->_dwriteGlyphRun.fontFace->GetDesignGlyphMetrics(run->_dwriteGlyphRun.glyphIndices,
+                                                                                 run->_dwriteGlyphRun.glyphCount,
+                                                                                 glyphMetrics,
+                                                                                 run->_dwriteGlyphRun.isSideways));
+
+            CTFontRef font = static_cast<CTFontRef>([run->_attributes objectForKey:static_cast<NSString*>(kCTFontAttributeName)]);
+            CGFloat pointSize = kCTFontSystemFontSize;
+            if (font) {
+                pointSize = CTFontGetSize(font);
+            }
+
+            DWRITE_FONT_METRICS fontMetrics;
+            run->_dwriteGlyphRun.fontFace->GetMetrics(&fontMetrics);
+
+            line->_ascent =
+                std::accumulate(glyphMetrics,
+                                glyphMetrics + run->_dwriteGlyphRun.glyphCount,
+                                line->_ascent,
+                                [&](CGFloat ascent, DWRITE_GLYPH_METRICS metrics) {
+                                    return std::max(ascent,
+                                                    _CoreTextScaleMetric(metrics.verticalOriginY, pointSize, fontMetrics.designUnitsPerEm));
```

Comment:
>pointSize, fontMetrics.designUnitsPerEm [](start = 98, length = 39)

from perf standpoint it would be better to compute pointSize/fontMetrics.designUnitsPerEm once and use it.